aleph/views/__init__.py

```diff
@@ -19,25 +22,44 @@
 from aleph.views.entitysets_api import blueprint as entitysets_api
 from aleph.views.exports_api import blueprint as exports_api
 
+from aleph.core import settings
+from aleph.views.util import get_authz
+
+
+def require_auth():
+    authz = get_authz(request)
+    if authz is None:
+        raise Unauthorized()
+
+
+def register_blueprint(app, blueprint, **kw):
+    if settings.REQUIRE_AUTH:
```

Comment:
My intuition would be that this wants to live on `authz` somewhere to make it more coherent with the rest of the system. Maybe like a `authz.can_browse_anonymous` that gets `require`'d when no other check is conducted (and is also implicitly checked by `can()` etc. - analogous to `session_write`? Doing it at the blueprint layer introduces a whole new dimension to authz that's sort of invisible and unexpected.

Reply:
Ok, in hindsight that's definitely a much cleaner implementation. Thanks @pudo!